packages/react-router/tests/useHoistedRoutes.test.tsx

```diff
@@ -0,0 +1,130 @@
+import { type RootRoute, type Route } from "../src/routeRegistry.ts";
+
+import { renderHook } from "@testing-library/react";
+import { type UseHoistedRoutesOptions, useHoistedRoutes } from "../src/useHoistedRoutes.ts";
+
+test("hoisted routes are at the root", () => {
+    const routes: RootRoute[] = [
+        { path: "/foo", element: <div>Foo</div> },
+        { path: "/hoisted", element: <div>Hoisted</div>, hoist: true }
+    ];
+
+    const wrapManagedRoutes = (managedRoutes: Route[]) => {
+        return {
+            path: "/wrapper",
+            children: [
+                ...managedRoutes
+            ]
+        };
+    };
+
+    const { result } = renderHook(({ routes: x, wrapManagedRoutes: y }) => useHoistedRoutes(x, y), {
+        initialProps: { routes, wrapManagedRoutes }
+    });
+
+    expect(result.current.length).toBe(2);
+    expect(result.current[0].path).toBe("/hoisted");
+});
+
+test("managed routes are wrapped", () => {
+    const routes: RootRoute[] = [
+        { path: "/foo", element: <div>Foo</div> },
+        { path: "/hoisted", element: <div>Hoisted</div>, hoist: true }
+    ];
+
+    const wrapManagedRoutes = (managedRoutes: Route[]) => {
+        return {
+            path: "/wrapper",
+            children: [
+                ...managedRoutes
+            ]
+        };
+    };
+
+    const { result } = renderHook(({ routes: x, wrapManagedRoutes: y }) => useHoistedRoutes(x, y), {
+        initialProps: {
+            routes,
+            wrapManagedRoutes
+        }
+    });
+
+    expect(result.current.length).toBe(2);
+    expect(result.current[1].path).toBe("/wrapper");
+    // eslint-disable-next-line testing-library/no-node-access
+    expect(result.current[1].children![0].path).toBe("/foo");
+});
+
+test("when a restricted route is hoisted, throw an error", () => {
+    // Prevent the expected exception from printing in the console.
+    jest.spyOn(console, "error").mockImplementation(jest.fn());
+
+    const routes: RootRoute[] = [
+        { path: "/foo", element: <div>Foo</div> },
+        { path: "/hoisted", element: <div>Hoisted</div>, hoist: true },
+        { path: "/bar", element: <div>Bar</div>, hoist: true }
+    ];
+
+    const wrapManagedRoutes = (managedRoutes: Route[]) => {
+        return {
+            path: "/wrapper",
+            children: [
+                ...managedRoutes
+            ]
+        };
+    };
+
+    expect(() => renderHook(({ routes: x, wrapManagedRoutes: y, options }) => useHoistedRoutes(x, y, options), {
+        initialProps: {
+            routes,
+            wrapManagedRoutes,
+            options: {
+                allowedPaths: [
+                    "/hoisted"
+                ]
+            } as UseHoistedRoutesOptions
```

Comment:
Hmm where does the is inferred from then?

Reply:
I relooked and your right, it's needed. 

However you should prefer the keyword "satisfies UseHoistedRoutesOptions" instead of as. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator